tests/integration_tests/security/test_jail.py

```diff
@@ -104,8 +107,66 @@ def test_arbitrary_usocket_location(test_microvm_with_initrd):
                 test_microvm.jailer.uid, test_microvm.jailer.gid)
 
 
-def check_cgroups(cgroups, cgroup_location, jailer_id):
-    """Assert that every cgroup in cgroups is correctly set."""
+@functools.lru_cache(maxsize=None)
+def cgroup_v1_available():
+    """Check if cgroup-v1 is disabled on the system."""
+    with open("/proc/cmdline") as cmdline_file:
+        cmdline = cmdline_file.readline()
+        return bool("cgroup_no_v1=all" not in cmdline)
+
+
+@pytest.fixture
+def sys_setup_cgroups():
```

Comment:
Can this be a helper function instead of a fixture ?

Reply:
It can be a helper function but why is a fixture not appropriate here?
The purpose of this code is to set-up the croups environment on the system so that the tests can run successfully.
Maybe a more suggestive name is needed for the fixture like `cgroups_env`. What do you think?

Reply:
One issue I have with fixtures is that it makes it harder to read and to reason about how the env is being setup. I would prefer to have it done imperatively in the test code vs declarative as you do with fixtures.